environment.go

```diff
@@ -467,6 +508,39 @@ func (e *envState) GetSender(key SenderKey) (send.Sender, error) {
 	return sender, nil
 }
 
+func (e *envState) RegisterCloser(name string, closer func(context.Context) error) {
+	e.mu.Lock()
+	defer e.mu.Unlock()
+
+	e.closers[name] = closer
```

Comment:
Add some logic to prevent overwriting a closer?

Reply:
I'm not super worried about this case. I've added a warning.